web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js

```diff
@@ -41,13 +36,18 @@ odoo.define('web_widget_char_switchcase', function (require) {
             switch (this.current_transformation) {
                 case 'upper':
                     // HELLO WORLD!
-                    return function (val) { return val.toUpperCase(); };
+                    return function (val) {
+                        if (!val) return val;
+                        return val.toUpperCase();};
                 case 'lower':
                     // hello world!
-                    return function (val) { return val.toLowerCase(); };
+                    return function (val) {
+                        if (!val) return val;
```

Comment:
you remove all these checks now

Reply:
Anyway, added the `check_val` in https://github.com/OCA/web/pull/1079/commits/63e2623b69d22a27b224fe80586e2fef4a9ac99e